src/toplevel.c

```diff
@@ -1112,8 +1121,11 @@ JL_DLLEXPORT jl_value_t *jl_toplevel_eval_in(jl_module_t *m, jl_value_t *ex)
     jl_value_t *v = NULL;
     int last_lineno = jl_lineno;
     const char *last_filename = jl_filename;
+    jl_task_t *ct = jl_current_task;
     jl_lineno = 1;
     jl_filename = "none";
+    size_t last_age = ct->world_age;
+    ct->world_age = jl_atomic_load_relaxed(&jl_world_counter);
     JL_TRY {
```

Comment:
JL_TRY won't know that you've mutated the world outside of its control
```suggestion
    JL_TRY {
        ct->world_age = jl_atomic_load_relaxed(&jl_world_counter);
```